pkg/lightning/backend/local/local.go

```diff
@@ -2152,41 +2160,52 @@ func sortAndMergeRanges(ranges []Range) []Range {
 }
 
 func filterOverlapRange(ranges []Range, finishedRanges []Range) []Range {
-	if len(finishedRanges) == 0 {
+	if len(ranges) == 0 || len(finishedRanges) == 0 {
```

Comment:
The current implementation should be correct. But it took me a long time to understand the logic 😂. 
I think these logic may  be simplified. I wrote a demo like this(not tested😂):
```go
func filterOverlapRange(ranges []Range, finishedRanges []Range) []Range {
	var result []Range
	for _, r := range ranges {
		start, end := r.Start, r.End
		for bytes.Compare(start, end) < 0 && len(finishedRanges) > 0 && bytes.Compare(finishedRanges[0].Start, end) < 0 {
			fr := finishedRanges[0]
			// No overlap, skip it.
			if bytes.Compare(fr.End, start) <= 0 {
				finishedRanges = finishedRanges[1:]
				continue
			}
			if bytes.Compare(start, fr.Start) < 0 {
				result = append(result, Range{start, fr.Start})
			}
			start = fr.End
			finishedRanges = finishedRanges[1:]
		}
		if bytes.Compare(start, end) < 0 {
			result = append(result, Range{start, end})
		}
	}
	return result
}
```

Reply:
+1